p2p/host/autorelay/autorelay.go

```diff
@@ -141,46 +142,83 @@ func (ar *AutoRelay) hostAddrs(addrs []ma.Multiaddr) []ma.Multiaddr {
 func (ar *AutoRelay) background(ctx context.Context) {
 	defer ar.refCount.Done()
 
-	subReachability, _ := ar.host.EventBus().Subscribe(new(event.EvtLocalReachabilityChanged))
+	subReachability, err := ar.host.EventBus().Subscribe(new(event.EvtLocalReachabilityChanged))
+	if err != nil {
+		log.Error("failed to subscribe to the EvtLocalReachabilityChanged")
+		return
+	}
 	defer subReachability.Close()
+	subConnectedness, err := ar.host.EventBus().Subscribe(new(event.EvtPeerConnectednessChanged))
+	if err != nil {
+		log.Error("failed to subscribe to the EvtPeerConnectednessChanged")
+		return
+	}
+	defer subConnectedness.Close()
 
 	ticker := time.NewTicker(rsvpRefreshInterval)
 	defer ticker.Stop()
 
-	// when true, we need to identify push
-	push := false
-
 	for {
+		// when true, we need to identify push
+		var push bool
+
 		select {
-		case ev, ok := <-subReachability.Out():
+		case ev, ok := <-subConnectedness.Out():
 			if !ok {
 				return
 			}
-			evt, ok := ev.(event.EvtLocalReachabilityChanged)
+			evt := ev.(event.EvtPeerConnectednessChanged)
+			switch evt.Connectedness {
+			case network.Connected:
+				// If we just connect to one of our static relays, get a reservation immediately.
```

Comment:
this might make discovery strategy refactoring tricky, but i guess it is fine for now.